app/models/concerns/best_bet.rb

```diff
@@ -30,7 +33,7 @@ def display_status
 
   def match_keyword_values_only_requires_keywords(keywords_collection)
     if keywords_collection.reject(&:marked_for_destruction?).blank? && match_keyword_values_only?
```

Comment:
Github is still possessed  - my comment disappeared again. Hopefully the 3rd time will be a charm...

A guard clause is preferred to a conditional here:
```ruby
  def match_keyword_values_only_requires_keywords(keywords_collection)
    return unless keywords_collection.reject(&:marked_for_destruction?).blank? &&
      match_keyword_values_only?

    errors.add(:match_keyword_values_only, 'requires at least one keyword')
  end
```